pkg/addons/client.go

```diff
@@ -240,8 +241,13 @@ func needsUpdate(existing, o *unstructured.Unstructured) bool {
 	// TODO: we should have tests that monitor these diffs:
 	// 1) when a cluster is created
 	// 2) when sync is run twice back-to-back on the same cluster
-	for _, diff := range deep.Equal(*existing, *o) {
-		log.Infof("- " + diff)
+
+	// Don't show a diff if kind is Secret
+	oGroupKind := o.GroupVersionKind().GroupKind()
+	if strings.ToLower(oGroupKind.String()) != "secret" {
```

Comment:
The simplest thing to do here in order to test this is to move this part into a  `printDiff(existing, o) bool` function that returns true when something is printed and false otherwise, then write tests for that function.

Reply:
thanks @kargakis you are right